### PR TITLE
Supporting OpenSSL

### PR DESCRIPTION
**Pull request by nmittler**
- Upgrading to Netty 4.1.0.Beta6
- enabling OpenSSL in negotiation codepaths

### CODE REVIEW

**Comment by nmittler** [comment]
@ejona86

File: netty/src/main/java/io/grpc/netty/ProtocolNegotiators.java [hunk 1]
```diff
@@ -73,7 +71,34 @@ private ProtocolNegotiators() {
...
+
+      @Override
+      public void userEventTriggered(ChannelHandlerContext ctx, Object evt) throws Exception {
+        if (evt instanceof SslHandshakeCompletionEvent) {
+          SslHandshakeCompletionEvent handshakeEvent = (SslHandshakeCompletionEvent) evt;
+          if (handshakeEvent.isSuccess()) {
+            SslHandler handler = ctx.pipeline().get(SslHandler.class);
+            if (handler.applicationProtocol() != null) {
+              // Successfully negotiated the protocol.
+              ctx.pipeline().remove(this);
```
Comment (ejona86) [inline netty/src/main/java/io/grpc/netty/ProtocolNegotiators.java]: Use `writeBufferedAndRemove()` or don't use `AbstractNegotiationHandler`. I don't think anything will notice the exceptions passed to failed, fyi.

Comment (ejona86) [inline netty/src/main/java/io/grpc/netty/ProtocolNegotiators.java]: I mixed up `AbstractNegotiationHandler` and `AbstractBufferingHandler`. Does anything actually notice the failure? I didn't think anything would write to the channel if the channel fails before negotiation completes.

Comment (nmittler) [inline netty/src/main/java/io/grpc/netty/ProtocolNegotiators.java]: As discussed offline ... server doesn't need to handle failure in the same way. I've reverted the breakout of `AbstractNegotiationHandler` and just handle the TLS upgrade event in the server case.

**Comment by nmittler** [comment]
@ejona86 PTAL

**Comment by ejona86** [comment]
@nmittler LGTM. Thank you!

**Comment by nmittler** [comment]
Cherry-picked as aefefb5

**Pull request closed without merging**